[DOC][master] #63469 Issue - Replaced the key rejection paragraph with suggested fix & Added a links..

v-aryan opened the pull request:
### What does this PR do?
This PR fixes the first suggested issue which is mentioned in the guides for replacing the content with another and adding a link for [removal/rejection sections]


### What issues does this PR fix or reference?
Fixes:
The Salt Master generates a new AES key to encrypt its publications at certain events such as a Master restart or the removal of a Minion key. If you are encountering this problem of too many minions re-authing against the Master, you will need to recalibrate your setup to reduce the rate of events like a Master restart or Minion key removal (salt-key -d).

To

The Salt Master generates a new AES key to encrypt its publications at certain events such as a Master restart or the removal/rejection of a Minion key. If you are encountering this problem of too many minions re-authing against the Master, you will need to recalibrate your setup to reduce the rate of events like a Master restart, Minion key removal (salt-key -d) or Minion key rejection (salt-key -r).

And

A rotating AES key is used to encrypt jobs that are sent to the Salt minion by the Salt master, and to encrypt connections to the Salt master fileserver. A new key is generated and used each time the Salt master restarts and each time a Salt minion key is deleted using the salt-key command.

To

A rotating AES key is used to encrypt jobs that are sent to the Salt minion by the Salt master, and to encrypt connections to the Salt master fileserver. A new key is generated and used each time the Salt master restarts and each time a Salt minion key is deleted/rejected using the salt-key command.



### Merge requirements satisfied?
First suggested fix was fixed and the second suggested fix cannot be find which is in "ROTATING SECURITY KEYS" later on it was mentioned that it can be founded on next repository ("https://github.com/saltstack/salt-get-started/blob/7d87b3323913e9b5191fee739d52894ac6ed3811/system/communication.md").

aryanvarma7272 commented:
Please tell the suggestive changes which are required , if something wrong ?

aryanvarma7272 commented:
Please do tell about the changes which are wrong and need to be solved !

dmurphy18 commented on doc/man/salt-key.1:
```diff
@@ -232,8 +232,9 @@ Accepts all pending keys.
-in addition to pending keys). Globs are supported.
+-r REJECT, --reject=REJECT
+Reject the specified public key (use --include-all to match accepted keys in addition to pending keys). Globs are supported.
+It may cause dramatic performance salt-msater degradation. See [performance guide](https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing) for details
```
spelling msater -> master

twangboy replied:
Fixed

dmurphy18 commented on doc/man/salt.7:
```diff
@@ -84419,8 +84419,12 @@ Accepts all pending keys.
+-r REJECT, --reject=REJECT
+Reject the specified public key (use --include-all to match accepted keys in addition
+ to pending keys). Globs are supported.
+It may cause dramatic performance salt-msater degradation. See [performance guide]
```
spelling msater -> master

dmurphy18 commented on doc/ref/cli/salt-key.rst:
```diff
@@ -102,8 +102,9 @@ Actions
-    in addition to pending keys). Globs are supported.
+    -r REJECT, --reject=REJECT
+Reject the specified public key (use --include-all to match accepted keys in addition to pending keys). Globs are supported.
+It may cause dramatic performance salt-msater degradation. See `performance guide <https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing>` for details
```
spelling msater -> master

dmurphy18 commented on salt/utils/parsers.py:
```diff
@@ -2516,9 +2516,12 @@ def _mixin_setup(self):
+                "-r REJECT, --reject=REJECT"
+                "Reject the specified public key (use --include-all to match" 
+                "accepted keys in addition to pending keys). Globs are supported."
+                "It may cause dramatic performance salt-msater degradation." 
```
spelling msater -> master

dmurphy18 reviewed (dismissed):
Spelling mistakes, and should the updated messages wrap at 72 column, or let them free flow and force the user to scroll to end of line, believe they need to wrap similar to the messages being replaced.

twangboy commented:
Please address the changes requested above

bdrx312 commented on pkg/common/fish-completions/salt-key.fish:
```diff
@@ -25,7 +25,7 @@ complete -c salt-key         -f -s p -l print                -d "Print the speci
 complete -c salt-key         -r      -l pub                  -d "The public-key file to create a signature for"
 complete -c salt-key         -f -s R -l reject-all           -d "Reject all pending keys"
-complete -c salt-key         -f -s r -l reject               -d "Reject the specified public key (use --include-all to match accepted keys in addition to pending keys).  Globs are supported."
+complete -c salt-key         -f -s r -l reject               -d "-r REJECT, --reject=REJECT Reject the specified public key (use --include-all to match accepted keys in addition to pending keys). Globs are supported.It may cause dramatic performance salt-msater degradation. See [performance guide](https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing) ...
```
`supported.It`: Add space between `supported.` and `It`

bdrx312 commented on salt/utils/parsers.py:
```diff
@@ -2516,9 +2516,12 @@ def _mixin_setup(self):
+                "accepted keys in addition to pending keys). Globs are supported."
+                "It may cause dramatic performance salt-master degradation."
+                "See [performance guide]"
+                "(https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing) for details"
```
Add a space to the beginning of each string line so that there is a space between words and sentences.

```python
                " accepted keys in addition to pending keys). Globs are supported."
                " It may cause dramatic performance salt-master degradation."
                " See [performance guide]"
                "(https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing) for details"
```

bdrx312 commented on salt/utils/parsers.py:
```diff
@@ -2516,9 +2516,12 @@ def _mixin_setup(self):
-                "Reject the specified public key. Use --include-accepted and "
-                "--include-denied to match accepted and denied keys in "
-                "addition to pending keys. Globs are supported."
+                "-r REJECT, --reject=REJECT"
```
None of the other options list the arguments in the help string like this. I suggest removing it from here to or add it to all the options. I am not very familiar with optparse since argparse basically replaced it ages ago, but with argparse the argument flags are automatically included in the usage statement; you do not manually include them in the help string. I could be wrong, but I assume that optparse has similar functionality.

dwoz committed:
Mention key rejection as AES key rotation trigger

Fixes #63469: adds rejection (salt-key -r) alongside deletion (salt-key -d)
as an event that causes the Salt master to rotate its AES key and triggers
minion re-authentication. Updates doc/ref/cli/salt-key.rst with a note
linking to the performance guide, wraps the long line in intro_scale.rst and
restores inline code formatting, fixes the fish-completion description, and
adds a plain-text performance note to the parsers.py help string.
